collectd/files/ping.conf

```diff
@@ -1,17 +1,30 @@
+{%- from "collectd/map.jinja" import collectd_settings with context %}
+{%- set ping_settings = collectd_settings.plugins.ping %}
+
+{%- set hfg = ping_settings.hosts_from_grains %}
+
 {% if hfg and salt['mine.get'](hfg.target, hfg.fun, hfg.expr_form) -%}
 {% set hosts = hosts + salt['mine.get'](hfg.target, hfg.fun, hfg.expr_form).values() -%}
 {% endif -%}
+
+#
+# DO NOT EDIT
+#
+# This file is managed by salt via {{ source }}
+# Modify the config that generates this file instead
+#
+
 LoadPlugin ping
+
 <Plugin "ping">
-  {%- for host in hosts %}
+  {%- for host in ping_settings.hosts %}
   Host "{{ host }}"
   {%- endfor %}
 
-
-  {{ 'Interval "{}"'.format(interval) if interval else '' }}
-  {{ 'Timeout "{}"'.format(timeout) if timeout else '' }}
-  {{ 'TTL "{}"'.format(ttl) if ttl else '' }}
-  {{ 'SourceAddress "{}"'.format(sourceaddress) if sourceaddress else '' }}
-  {{ 'Device "{}"'.format(device) if device else '' }}
-  {{ 'MaxMissed "{}"'.format(maxmissed) if maxmissed else '' }}
+  Interval "{{ ping_settings.interval }}"
+  Timeout "{{ ping_settings.timeout }}"
+  TTL "{{ ping_settings.ttl }}"
+  SourceAddress "{{ ping_settings.sourceaddress }}"
+  Device "{{ ping_settings.device }}"
+  MaxMissed "{{ ping_settings.maxmissed }}"
```

Comment:
More "If you are going to take out the ability to fall back on collectd defaults here, you should fix the defaults in the map.jinja file."